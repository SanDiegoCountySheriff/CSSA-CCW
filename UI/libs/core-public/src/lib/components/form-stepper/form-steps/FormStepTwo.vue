<template>
  <div>
    <v-form
      ref="form"
      v-model="valid"
    >
      <v-subheader class="sub-header font-weight-bold">
        {{ $t('Id Information') }}
      </v-subheader>
      <v-row>
        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <v-text-field
            v-model="id.idNumber"
            :label="$t('Id number')"
            :rules="[v => !!v || 'Id  number is required']"
            required
          />
        </v-col>

        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <v-text-field
            :label="$t(' Issuing State')"
            :rules="[v => !!v || 'Issuing state is required']"
            v-model="id.issuingState"
          />
        </v-col>
      </v-row>

      <v-divider />
      <v-subheader class="sub-header font-weight-bold">
        {{ $t(' Date of birth') }}
      </v-subheader>

      <v-row>
        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <v-date-picker
            v-model="DOBInfo.DOB"
            label="Date of birth"
          />
          <v-alert
            dense
            outlined
            type="error"
            v-if="!DOBInfo.DOB"
          >
            {{ $t('Date of birth cannot be blank!') }}
          </v-alert>
        </v-col>

        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <v-text-field
            :label="$t('Current Age')"
            :rules="[v => !!v || 'Current age is required']"
            v-model="DOBInfo.currentAge"
          />
          <v-text-field
            :label="$t('Birth city')"
            :rules="[v => !!v || 'Birth city cannot be blank']"
            v-model="DOBInfo.birthCity"
          />
          <v-text-field
            :label="$t('Birth state')"
            :rules="[v => !!v || 'Birth state cannot be blank']"
            v-model="DOBInfo.birthState"
          />
          <v-text-field
            :label="$t('Birth country')"
            :rules="[v => !!v || 'Birth country cannot be blank']"
            v-model="DOBInfo.birthCountry"
          />
        </v-col>
      </v-row>

      <v-divider />
      <v-subheader class="sub-header font-weight-bold">
        {{ $t('Citizenship Information') }}
      </v-subheader>

      <v-row>
        <v-col
          cols="6"
          md="5"
          sm="3"
          class="pl-5"
        >
          <RadioGroupInput
            :label="'Citizen'"
            :layout="'row'"
            :options="[
              { label: 'Yes', value: true },
              { label: 'No', value: false },
            ]"
            @input="
              v => {
                citizenshipInfo.citizen = v;
              }
            "
          />
          <v-alert
            dense
            outlined
            type="error"
            v-if="!citizenshipInfo.citizen"
          >
            {{ $t('Must select Yes or No!') }}
          </v-alert>
        </v-col>

        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <v-select
            v-model="citizenshipInfo.militaryStatus"
            :items="items"
            :label="$t('Military Status')"
          />

          <v-alert
            dense
            outlined
            type="error"
            v-if="!citizenshipInfo.militaryStatus"
          >
            {{ $t('Must select a status') }}
          </v-alert>
        </v-col>
      </v-row>
    </v-form>
    <v-divider />
    <FormButtonContainer
      :valid="valid"
      @submit="handleSubmit"
    />
  </div>
</template>

<script setup lang="ts">
import { reactive, ref } from 'vue';
import { useIdStore } from '@shared-ui/stores/id';
import { useDOBStore } from '@shared-ui/stores/DOB';
import { useCitizenshipStore } from '@shared-ui/stores/citizenship';
import RadioGroupInput from '@shared-ui/components/inputs/RadioGroupInput.vue';
import FormButtonContainer from '@core-public/components/containers/FormButtonContainer.vue';
import { CitizenshipType, DOBType, IdType } from '@shared-utils/types/defaultTypes';

interface FormStepOneProps {
  handleNextSection: () => void;
}

const props = withDefaults(defineProps<FormStepOneProps>(), {
  handleNextSection: () => null,
});

const id = reactive({} as IdType);
const DOBInfo = reactive({} as DOBType);
const citizenshipInfo = reactive({} as CitizenshipType);
const items = ref(['Active', 'Reserve', 'Discharged', 'Retired', 'N/A']);
const valid = ref(false);

const idStore = useIdStore();
const DOBStore = useDOBStore();
const citizenshipStore = useCitizenshipStore();

function handleSubmit() {
  idStore.setId(id);
  DOBStore.setDOBInfo(DOBInfo);
  citizenshipStore.setCitizenshipInfo(citizenshipInfo);
  props.handleNextSection();
}
</script>
