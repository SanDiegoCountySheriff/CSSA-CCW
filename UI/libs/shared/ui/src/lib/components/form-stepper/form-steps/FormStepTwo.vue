<template>
  <div>
    <v-form>
      <v-subheader class="subHeader font-weight-bold"> Id Information </v-subheader>
      <v-row>
        <v-col cols="6" md="5" sm="3">
          <v-text-field
            v-model="id.idNumber"
            label="Id number"
            :rules="[v => !!v || 'Id  number is required']"
            required
          />
        </v-col>

        <v-col cols="6" md="5" sm="3">
          <v-text-field
            v-model="id.issuingState"
            label=" Issuing State"
            :rules="[v => !!v || 'Issuing state is required']"
            required
          />
        </v-col>
      </v-row>
      <v-divider />
      <v-subheader class="subHeader font-weight-bold"> Date of birth </v-subheader>

      <v-row>
        <v-col cols="6" md="5" sm="3">
          <v-date-picker v-model="DOBInfo.DOB" label="Date of birth" />
          <!-- TODO: need a error block here if the data picker is invalid -->
        </v-col>
        <v-col cols="6" md="5" sm="3">
          <v-text-field v-model="DOBInfo.birthCity" label="Birth city" required />
          <v-text-field v-model="DOBInfo.birthState" label="Birth state" />
          <v-text-field v-model="DOBInfo.birthCountry" label="Birth country" required />
        </v-col>
      </v-row>

      <v-divider />
      <v-subheader class="subHeader font-weight-bold"> Citizenship Information </v-subheader>
      <v-row>
        <v-col cols="6" md="5" sm="3">
          <v-radio-group class="ml-5" v-model="citizenshipInfo.citizen" label="Citizen">
            <v-radio label="Yes" :value="true" />

            <v-radio label="No" :value="false" />
          </v-radio-group>
        </v-col>
        <v-col cols="6" md="5" sm="3">
          <v-select v-model="citizenshipInfo.militaryStatus" :items="items" label="Military Status"> </v-select>
        </v-col>
      </v-row>
    </v-form>
    <v-divider />
    <div class="form-btn-container">
      <v-btn color="secondary mr-2" @click="handleSubmit"> Continue </v-btn>
      <!-- TODO: this needs to save to save the current state to local storage or call the api to save in
      the db
       -->
      <v-btn color="info mr-2">Save and Exit</v-btn>
      <!-- TODO: Make this return to the home page with out saving the form at all -->
      <v-btn color="error mr-2"> Cancel</v-btn>
    </div>
    <FormErrorAlert v-if="errors.length > 0" :errors="errors" />
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue';
import { Citizenship, DOB, Id } from '@shared-ui/types/defualtTypes';
import { validateForm } from '@shared-ui/validators/form-validators/formStepTwoValidators';
import FormErrorAlert from '@shared-ui/components/alerts/FormErrorAlert.vue';
import { addIdAction } from '@shared-ui/helpers/storeHelpers/idActions';
import { addCitizenshipAction } from '@shared-ui/helpers/storeHelpers/citizenActions';
import { addDOBAction } from '@shared-ui/helpers/storeHelpers/DOBActions';

export default defineComponent({
  name: 'FormStepTwo',
  components: { FormErrorAlert },
  props: {
    handleNextSection: {
      type: Function as PropType<() => void>,
      default: () => null,
    },
  },
  data() {
    return {
      id: {} as Id,
      DOBInfo: {} as DOB,
      citizenshipInfo: {} as Citizenship,
      items: ['Active', 'Reserve', 'Discharged', 'Retired', 'N/A'],
      errors: [] as Array<string>,
    };
  },
  methods: {
    handleSubmit() {
      const formData = {
        id: this.id,
        dob: this.DOBInfo,
        citizenship: this.citizenshipInfo,
      };
      this.errors = validateForm(formData);
      if (this.errors.length <= 0) {
        addIdAction(this.id);
        addDOBAction(this.DOBInfo);
        addCitizenshipAction(this.citizenshipInfo);
        this.handleNextSection();
      }
    },
  },
});
</script>

<style scoped>
.subHeader {
  font-size: 1.5rem;
}

.form-btn-container {
  width: 90%;
  margin-top: 0.5rem;
  display: flex;
  justify-content: flex-end;
}
</style>
