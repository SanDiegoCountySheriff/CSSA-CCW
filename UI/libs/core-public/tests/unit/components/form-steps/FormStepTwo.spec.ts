import FormStepTwo from '@core-public/components/form-stepper/form-steps/FormStepTwo.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

const pinia = createTestingPinia();
const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('FormStepTwo', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    wrapper = mount(FormStepTwo, {
      localVue,
      vuetify,
      pinia,
      mocks: tMock,
      propsData: {
        handleNextSection: () => null,
        handlePreviousSection: () => null,
      },
      data() {
        return {
          DOBInfo: {
            DOB: '',
          },
        };
      },
    });
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render all the text inputs', () => {
    const textFields = wrapper.findAllComponents('.v-text-field');

    expect(textFields).toHaveLength(7);
  });
});
