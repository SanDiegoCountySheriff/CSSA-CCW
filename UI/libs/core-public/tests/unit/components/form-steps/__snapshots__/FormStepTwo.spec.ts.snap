// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormStepTwo should match the snapshot 1`] = `
<div>
  <form novalidate="novalidate" class="v-form">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      Id Information
    </div>
    <div class="row">
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-asterisk theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-5" class="v-label theme--light" style="left: 0px; position: absolute;">Id number</label><input id="input-5" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-asterisk theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-9" class="v-label theme--light" style="left: 0px; position: absolute;"> Issuing State</label><input id="input-9" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      Date of birth
    </div>
    <div class="row">
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-picker v-card v-picker--date theme--light" label="Date of birth">
          <div class="v-picker__title primary">
            <div class="v-date-picker-title">
              <div class="v-picker__title__btn v-date-picker-title__year">2022</div>
              <div class="v-picker__title__btn v-date-picker-title__date v-picker__title__btn--active">
                <transition-stub name="picker-transition">
                  <div>&nbsp;</div>
                </transition-stub>
              </div>
            </div>
          </div>
          <div class="v-picker__body theme--light" style="width: 290px;">
            <transition-stub name="fade-transition">
              <div>
                <div class="v-date-picker-header theme--light"><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default" aria-label="Previous month"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button>
                  <div class="v-date-picker-header__value">
                    <transition-stub name="tab-transition">
                      <div class="accent--text"><button type="button">November 2022</button></div>
                    </transition-stub>
                  </div><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default" aria-label="Next month"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button>
                </div>
                <div class="v-date-picker-table v-date-picker-table--date theme--light">
                  <transition-stub name="tab-transition">
                    <table>
                      <thead>
                        <tr>
                          <th>S</th>
                          <th>M</th>
                          <th>T</th>
                          <th>W</th>
                          <th>T</th>
                          <th>F</th>
                          <th>S</th>
                        </tr>
                      </thead>
                      <tbody>
                        <tr>
                          <td></td>
                          <td></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">1</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">2</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-date-picker-table__current v-btn--rounded v-btn--outlined theme--light accent--text">
                              <div class="v-btn__content">3</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">4</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">5</div>
                            </button></td>
                        </tr>
                        <tr>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">6</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">7</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">8</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">9</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">10</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">11</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">12</div>
                            </button></td>
                        </tr>
                        <tr>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">13</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">14</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">15</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">16</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">17</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">18</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">19</div>
                            </button></td>
                        </tr>
                        <tr>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">20</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">21</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">22</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">23</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">24</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">25</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">26</div>
                            </button></td>
                        </tr>
                        <tr>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">27</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">28</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">29</div>
                            </button></td>
                          <td><button type="button" class="v-btn v-btn--text v-btn--rounded theme--light">
                              <div class="v-btn__content">30</div>
                            </button></td>
                          <td></td>
                          <td></td>
                          <td></td>
                        </tr>
                      </tbody>
                    </table>
                  </transition-stub>
                </div>
              </div>
            </transition-stub>
          </div>
        </div>
        <div role="alert" class="v-alert v-sheet v-sheet--outlined theme--light v-alert--dense v-alert--outlined error--text">
          <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon mdi mdi-alert theme--light error--text"></i>
            <div class="v-alert__content">
              Date of birth cannot be blank!
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-asterisk theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-30" class="v-label theme--light" style="left: 0px; position: absolute;">Birth city</label><input id="input-30" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
        <div class="v-input theme--light v-text-field">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-asterisk theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-34" class="v-label theme--light" style="left: 0px; position: absolute;">Birth state</label><input id="input-34" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
        <div class="v-input theme--light v-text-field">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-asterisk theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-38" class="v-label theme--light" style="left: 0px; position: absolute;">Birth country</label><input id="input-38" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      Citizenship Information
    </div>
    <div class="row">
      <div class="pl-5 col-sm-3 col-md-5 col-6">
        <div class="radio-group">
          <div class="v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--row">
            <div class="v-input__control">
              <div class="v-input__slot" style="height: auto;">
                <div role="radiogroup" aria-labelledby="input-45" class="v-input--radio-group__input">
                  <legend id="input-45" class="v-label theme--light" style="left: 0px; position: relative;">Citizen</legend>
                  <div class="v-radio radio-group text--black theme--light">
                    <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-46" role="radio" type="radio" name="radio-45" value="true">
                      <div class="v-input--selection-controls__ripple"></div>
                    </div><label for="input-46" class="v-label theme--light" style="left: 0px; position: relative;">Yes</label>
                  </div>
                  <div class="v-radio radio-group text--black theme--light">
                    <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-48" role="radio" type="radio" name="radio-45" value="false">
                      <div class="v-input--selection-controls__ripple"></div>
                    </div><label for="input-48" class="v-label theme--light" style="left: 0px; position: relative;">No</label>
                  </div>
                </div>
              </div>
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
          <!---->
        </div>
        <div role="alert" class="v-alert v-sheet v-sheet--outlined theme--light v-alert--dense v-alert--outlined error--text">
          <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon mdi mdi-alert theme--light error--text"></i>
            <div class="v-alert__content">
              Must select Yes or No!
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field v-select">
          <div class="v-input__control">
            <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-54" class="v-input__slot">
              <div class="v-select__slot"><label for="input-54" class="v-label theme--light" style="left: 0px; position: absolute;">Military Status</label>
                <div class="v-select__selections"><input id="input-54" readonly="readonly" type="text" aria-readonly="false" autocomplete="off"></div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
        <div role="alert" class="v-alert v-sheet v-sheet--outlined theme--light v-alert--dense v-alert--outlined error--text">
          <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon mdi mdi-alert theme--light error--text"></i>
            <div class="v-alert__content">
              Must select a status
            </div>
          </div>
        </div>
        <!---->
      </div>
    </div>
    <div class="container container--fluid">
      <hr role="separator" aria-orientation="horizontal" class="mb-3 v-divider theme--light">
      <div class="v-subheader sub-header font-weight-bold theme--light">
        Immigrant Information
      </div>
      <div class="row ml-5">
        <div class="col">
          <div class="v-input theme--light v-text-field">
            <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-asterisk theme--light error--text" style="font-size: 12px;"></i></div>
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-text-field__slot"><label for="input-63" class="v-label theme--light" style="left: 0px; position: absolute;">Country of Citizenship</label><input id="input-63" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="radio-group">
            <div class="v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--row">
              <div class="v-input__control">
                <div class="v-input__slot" style="height: auto;">
                  <div role="radiogroup" aria-labelledby="input-68" class="v-input--radio-group__input">
                    <legend id="input-68" class="v-label theme--light" style="left: 0px; position: relative;">Immigrant Alien</legend>
                    <div class="v-radio radio-group text--black theme--light">
                      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-69" role="radio" type="radio" name="radio-68" value="true">
                        <div class="v-input--selection-controls__ripple"></div>
                      </div><label for="input-69" class="v-label theme--light" style="left: 0px; position: relative;">Yes</label>
                    </div>
                    <div class="v-radio radio-group text--black theme--light">
                      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-71" role="radio" type="radio" name="radio-68" value="false">
                        <div class="v-input--selection-controls__ripple"></div>
                      </div><label for="input-71" class="v-label theme--light" style="left: 0px; position: relative;">No</label>
                    </div>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
            <!---->
          </div>
          <div class="radio-group">
            <div class="v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--row">
              <div class="v-input__control">
                <div class="v-input__slot" style="height: auto;">
                  <div role="radiogroup" aria-labelledby="input-76" class="v-input--radio-group__input">
                    <legend id="input-76" class="v-label theme--light" style="left: 0px; position: relative;">Non-Immigrant Alien</legend>
                    <div class="v-radio radio-group text--black theme--light">
                      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-77" role="radio" type="radio" name="radio-76" value="true">
                        <div class="v-input--selection-controls__ripple"></div>
                      </div><label for="input-77" class="v-label theme--light" style="left: 0px; position: relative;">Yes</label>
                    </div>
                    <div class="v-radio radio-group text--black theme--light">
                      <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-79" role="radio" type="radio" name="radio-76" value="false">
                        <div class="v-input--selection-controls__ripple"></div>
                      </div><label for="input-79" class="v-label theme--light" style="left: 0px; position: relative;">No</label>
                    </div>
                  </div>
                </div>
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
            <!---->
          </div>
        </div>
        <div class="col">
          <div class="v-input theme--light v-text-field">
            <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-asterisk theme--light error--text" style="font-size: 12px;"></i></div>
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-text-field__slot"><label for="input-83" class="v-label theme--light" style="left: 0px; position: absolute;">Country of Birth</label><input id="input-83" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
  <div class="form-btn-container"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
    Continue
  </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default secondary mx-2"><span class="v-btn__content">
    Save and Exit
  </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default error mr-2"><span class="v-btn__content">
    Cancel
  </span></button></div>
</div>
`;
